### PR TITLE
Feat/docs init structure and landingpage

### LINKED ISSUE #316: [Documentation] Landingpage
### Relevant documentation

Other

### Problem description (optional)

_No response_

### Desired solution

Landingpage for RefArch documentation:
- Description for what the refarch is used and what the advantages using it are
- Links to relevant entry points in documentation

### Additional context (optional)

_No response_

### No duplicate

- [X] I confirm that this issue is not a duplicate

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Init docs structure and init landing page.

**Reference**

Issues closes #316 

## Release Notes

- **New Features**
	- Enhanced documentation structure with new sections for "Development", "API-Gateway", "Integrations", and "S3-Integration".
	- Introduced a custom theme with a new logo in the navigation bar.
	- Updated hero section with a new project name and description.

- **Bug Fixes**
	- Improved formatting process by excluding `dist` and `node_modules` from Prettier.

- **Documentation**
	- Added various new sections to improve clarity and organization in the documentation.

- **Chores**
	- Updated dependencies in the project configuration.

### CODE REVIEW

File: docs/.vitepress/config.mts [hunk 1]
```diff
@@ -1,32 +1,65 @@
...
+      text: "View this page on GitHub",
+    },
     search: {
-      provider: 'local'
-    }
-  }
+      provider: "local",
+    },
+    footer: {
+      message: `<a href="https://opensource.muenchen.de/impress.html">Impress and Contact</a>`,
```
Comment (simonhir) [inline docs/.vitepress/config.mts:60]: @klml you know if that's ok or we would need to provide and maintain it ourself?

File: docs/.vitepress/config.mts [hunk 2]
```diff
@@ -1,32 +1,65 @@
...
+        ],
+      },
     ],
     sidebar: [
+      { text: "Overview", link: "/overview" },
+      { text: "API-Gateway", link: "/gateway" },
+      {
+        text: "Integrations",
+        link: "/integrations",
+        items: [{ text: "S3", link: "/integrations/s3" }],
```
Comment (devtobi) [inline docs/.vitepress/config.mts:38]: Why is only S3 included as a subitem?

Comment (simonhir) [reply docs/.vitepress/config.mts:38]: Just added the first one as boilerplate

**Review by devtobi — APPROVED** [review]
LGTM

File: docs/.vitepress/config.mts (hunk 1 above)
Comment (klml) [reply docs/.vitepress/config.mts:60]: @simonhir that's great, and also better than having your own imprint because then you don't have to maintain it

**Pull request merged**